src/LaravelDataTables/Providers/ServiceProvider.php

```diff
@@ -24,14 +30,52 @@ public function boot()
             $dataTable->setConfig($configuration);
             $request = Request::duplicate();
 
-            if ($request->get('csv') == 'prepare') {
+            if (in_array($request->get('csv'), ['prepare', 'email'])) {
                 $request->merge([
                     'start' => null,
                     'length' => null,
                 ]);
 
-                $csv = $dataTable->setRequest($request)
-                    ->makeResponseCsv();
+                $dataTable->setRequest($request);
+
+                if($request->get('csv') == 'email') {
+                    $user = Auth::user();
+                    $params = [
+                        'userId' => $user->getId(),
+                        'configuration' => $configuration,
+                        'filters' => $request->all(),
+                        'title' => $dataTable->getConfig()->getTitle() ??
+                            preg_replace('|^(.*?)DataTable$|s', '$1', basename(get_class($configuration))),
+                    ];
+
+                    dispatch(function () use ($params) {
+                        //TODO: discuss on resource limits
+                        // & error handling when exceeded
+                        ini_set('memory_limit','2G');
+                        set_time_limit(20*60);
+
+                        $dataTable = new DataTables;
+                        $dataTable->setConfig($params['configuration']);
+                        $request = (new \Illuminate\Http\Request())->merge($params['filters']);
+                        $request->merge([
+                            'start' => null,
+                            'length' => null,
+                        ]);
+
+                        $dataTable->setRequest($request);
+                        $csvData = $dataTable->makeResponseCsv();
+
+                        Mail::send(new DataTablesCsvEmail($params['userId'], $params['title'], $csvData));
```

Comment:
Shouldn't import a class from "App" namespace as this is definitively referencing the cloud-connect repo.

Instead use the IoC Container and a Contract.

app()->make(DataTablesCsvEmailContract::class, [ ... construct params ...])

and in the cloud-connect project, add this to the IoC in the AppServiceProvider

$this->app->bind(
    DataTablesCsvEmailContract::class, // from the datatables package namespace
    DataTablesCsvEmail::class // from App namespace in cloud-connect
);

Reply:
The correct way to do this would be to publish a config file for the package and have it set there. Defaulting to a Mailable which is included in the package, rather than having to add something specifically to the app service provider.

Reply:
Used the approach @russellgalpin suggested.